src/middlewared/middlewared/utils/__init__.py

```diff
@@ -111,8 +111,33 @@ def get(obj, path):
     return cur
 
 
+def casefold(obj):
+    if obj is None:
+        return None
+
+    if isinstance(obj, str):
+        return obj.casefold()
+
+    if isinstance(obj, (list, tuple)):
+        return [x.casefold() for x in obj]
+
+    raise ValueError(f'{type(obj)}: support for casefolding object type not implemented.')
+
+
 class filters(object):
-    opmap = {
+    case_opmap = {
```

Comment:
Can we just use operators from `opmap` but run `casefold` against arguments first? That way we won't have duplicate code.